tests/test_matrix_utils_connect_credentials.py

```
@@ -0,0 +1,308 @@
+            new=AsyncMock(return_value=None),
+        ),
+        patch("mmrelay.matrix_utils.logger") as mock_logger,
+    ):
+        result = await connect_matrix(config)
+
+    assert result is None
+    mock_logger.error.assert_called_with("Failed to load newly created credentials")
+
+
+@pytest.mark.asyncio
+async def test_connect_matrix_auto_login_failure(monkeypatch):
```

Ruff (0.15.9)

[warning] Unused function argument: `monkeypatch`

(ARG001)